docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,289 @@
+
+## New System Contract
+
+### Test Vectors
+
+- [ ] Call contexts
+    - [ ] Normal call to system contract from contract
+    - [ ] Delegate call to system contract from contract
+    - [ ] Static call to system contract from contract
+    - [ ] Code call to system contract from contract
+    - [ ] System contract as transaction entry-point
+    - [ ] Call from initcode
+        - [ ] Contract creating transaction
+        - [ ] Contract creating opcode
+- [ ] Inputs
+    - [ ] Verify combinations of valid inputs to the system contract
+        - [ ] Verify interesting edge values given the system contract functionality.
+    - [ ] Verify all zeros input.
+    - [ ] Verify 2^N-1 where N is a single or multiple valid bit-lengths.
+- [ ] Out-of-bounds checks
+    - [ ] Verify if the system contract has out-of-bounds conditions in its inputs and verify:
+        - [ ] Max value for each input
+        - [ ] Max value + 1 for each input
+- [ ] Input Lengths
+    - [ ] Zero-length calldata.
+    - [ ] System contract has static-length input
+        - [ ] Correct static-length calldata
+        - [ ] Calldata too short, where the value represents a correct but truncated input to the contract.
+        - [ ] Calldata too long, where the value represents a correct input to the contract with padded zeros.
+    - [ ] System contract has dynamic-length input
+        - [ ] Verify correct System contract execution for valid lengths, given different inputs.
+        - [ ] Calldata too short, given different inputs, where the value represents a correct but truncated input to the contract.
+        - [ ] Calldata too long, given different inputs, where the value represents a correct input to the contract with padded zeros.
+- [ ] Excessive Gas Cases
+    - [ ] If possible, simulate a scenario where the execution of the contract at the end of the block execution by the system address would result in excessive gas usage (100 million gas or more).
+    - [ ] Verify spending all block gas in calls to system contract (100 million gas or more).
+- [ ] System Contract Deployment
+    - [ ] Verify block execution behavior after fork activation if the system contract has not been deployed.
+    - [ ] Verify deployment transaction results in the system contract being deployed to the expected address.
+- [ ] Contract Variations
+    - [ ] Verify execution of the different variations of the contract for different networks (if any) results in the expected behavior
+- [ ] Contract Substitution
+    - [ ] Substitute the contract to modify its behavior when called by the system address (at the end of the block execution):
+        - [ ] Modified return value lengths
+        - [ ] Modify emitted logs
+- [ ] Fork transition
+    - [ ] Verify calling the system contract before its activation fork results in correct behavior (depends on the system contract implementation).
```

Comment:
Also good test, this should test for current EIPs that the system call is not done before the fork (but when the contract exists)

Reply:
Yes, we cannot ban calls to the system contract, so basically a test that is heavily dependent on the specific contract behavior needs to be written.